src/types.ts

```diff
@@ -155,9 +193,22 @@ export class TupleType extends ListOfTypes implements Type {
     accept<T>(visitor: TypeVisitor<T>): T {
         return visitor.visitTuple(this)
     }
+
+    equal(other: Type): boolean {
+        return other instanceof TupleType && super.equalSameOrder(other);
+    }
 }
 
 export class EnumType implements Type {
+    static Of(spec: { [name: string]: string | number }): EnumType {
```

Comment:
I added tihs, but are we actually using it here? either use it or remove it. keep it small

Reply:
addressed in 1103270